components/core/tests/test-NetworkReader.cpp

```diff
@@ -97,6 +100,23 @@ auto assert_curl_error_code(CURLcode expected, clp::NetworkReader const& reader)
     WARN(message_to_log);
     return false;
 }
+
+auto test_illegal_header(std::string const& header_name, std::string const& header_value) -> bool {
```

Comment:
1. For any helper functions inside anon namespace, we should declare the function prototype at the top and then implementations at the bottom.
2. We should also have a doc string explaining what this function does and what are the input/output.
3. We should use `std::string_view` instead of `std::string const&` whenever possible in C++>=17
4. We should add `[[nodiscard]]` attribute for functions that doesn't return `void` in their declaration.